src/hotspot/share/code/codeBlob.hpp

```diff
@@ -454,6 +458,25 @@ class AdapterBlob: public BufferBlob {
   virtual bool is_adapter_blob() const { return true; }
 };
 
+//----------------------------------------------------------------------------------------------------
+// MethodHandleIntrinsicBlob: used to hold MethodHandle intrinsic
+
+class MethodHandleIntrinsicBlob : public BufferBlob {
+private:
+  MethodHandleIntrinsicBlob(Method* method, int mhi_size, CodeBuffer *code_buffer);
+
+  Method*   _method;
```

Comment:
Sorry, I know that this is still a draft (I was taking a sneak peak), but note that `BufferBlob` subclasses have a fixed header size `sizeof(BufferBlob)` (see the `BufferBlob` constructor in codeBlob.cpp).

Adding a field here can lead to the the start of the code section being overwritten. Just FYI, in case you were encountering any weird bugs.

I think the right move, if you want to add a field here, is to extend `RuntimeBlob` directly, and pass the right header size to the super constructor (or pass the right header size from all `BufferBlob` subclasses).